lib/evmone/eof.cpp

```diff
@@ -186,7 +189,12 @@ EOFValidationError validate_eof(evmc_revision rev, bytes_view container) noexcep
     {
         if (rev < EVMC_CANCUN)
             return EOFValidationError::eof_version_unknown;
-        return validate_eof1(rev, container).second;
+
+        const auto header_or_error = validate_eof1(rev, container);
```

Comment:
It seems this function isn't even using the returned `EOF1Header` and so the entire deal with `pair`/`variant` seems useless.

Reply:
For simplicity we can change `validate_eof1()`  to return only error.

Reply:
We can also make pubic `validate_eof()` return variant, in general it seems useful to be able to validate and parse in one go.

But currently I found that it could be useful only in `eofparse` (which does `validate_eof()` + `read_valid_eof1_header()` combo)

State test runner seems to do these two operations in different places.